dataland-frontend/src/components/pages/CompanyCockpitPage.vue

```diff
@@ -38,7 +38,6 @@ import CompanyInfoSheet from "@/components/general/CompanyInfoSheet.vue";
 import { ARRAY_OF_FRAMEWORKS_WITH_VIEW_PAGE } from "@/utils/Constants";
 import ClaimOwnershipPanel from "@/components/resources/companyCockpit/ClaimOwnershipPanel.vue";
 import { getUserId } from "@/utils/KeycloakUtils";
-import { getErrorMessage } from "@/utils/ErrorMessageUtils";
 
```

Comment:
Wieso wird 9 mal gecheckt ob ich data-owner bin wenn ich auf die cockpit Seite gehe?:

![image](https://github.com/d-fine/Dataland/assets/99255296/328048fb-f8cb-4f95-89c1-2d513107be28)

Reply:
Vermutlich weil innerhalb der FrameworkSummaryPanel Komponente auf DataOwner geprüft wird, um die entsprechenden Buttons anzuzeigen.

Reply:
Ok aber das macht finde ich wenig Sinn.

Wieso prüft das cockpit nicht einfach einmal ob der user data-owner ist, und sagt den ganzen Panels einfach über eine prop Bescheid?

V.a. wächst das PRoblem mit jedem weiteren Framework das wir einführen..

Reply:
Es folgt dabei aber der gleichen Logik wie der check auf die Uploader Rolle

Reply:
Ehrlicherweise weiß ich nicht wo die Error logs herkommen. Ich habe zum Test alle neueingeführten console.log(error) statements entfernt und erhalte es trotzdem noch.

Reply:
Hab eine Idee wo es her kommt, aber ehrlicherweise verstehe ich nciht warum es passiert...

Reply:
Sollte durch das chaining weg sein

Reply:
Das chaining hat keinen Einfluss darauf, wie oft auf data-ownership geprüft wird.

Ich habe es lokal ausgeführt und es werden immer noch 9 requests geschossen.

Die Strategie mit der prop müsste umgesetzt werden, um das zu reparieren.

Reply:
Ich bezog mich auch auf die console errors. Das andere probiere ich gerade aus, das funktioniert aber noch nicht

Reply:
So, jetzt ist die Logik umgezogen in das Company Cockpit

Reply:
Jetzt hast du aber den Bug eingebaut, dass ich keine Provide-Data-Buttons sehe wenn ich als Data-Uploader mich einlogge.

Und die Variable `isUserAllowedToUpload` die du eingeführt hast, wird in der Komponente irgendwie gar nicht benutzt.

Reply:
Korrigiert, das war ein Bug

Reply:
Kurzer call. Mir ist etwas aufgefallen (nicht so shclimm, keine Sorge 😆 )